schemas/generate_json_schema.py

```diff
@@ -92,42 +87,51 @@ def iterate_models():
     top_model: BaseModel = create_model(
         "_TopModel_", **{m.__name__: (m, ...) for m in models}
     )
-    top_model.model_config["extra"] = "forbid"
-    top_model.model_config["json_schema_extra"] = staticmethod(clean_schema)
 
     schema: dict = top_model.model_json_schema(mode="serialization")
 
     for d in schema.get("$defs", {}).values():
         clean_schema(d)
 
-    return json.dumps(schema)
+    return schema
 
 
-def main(output):
+def main(*, ts_output_path: Path):
     json_schema = iterate_models()
-    schema_dir = mkdtemp()
-    schema_file_path = os.path.join(schema_dir, "schema.json")
 
-    with open(schema_file_path, "w") as f:
-        f.write(json_schema)
+    with TemporaryDirectory() as tmp_dir:
+        tmp_dir = Path(tmp_dir)
+        schema_file_path = tmp_dir / "schema.json"
 
-    json2ts_cmd = "yarn json2ts"
-    json2ts_exit_code = os.system(
-        f'{json2ts_cmd} -i {schema_file_path} -o {output} --bannerComment ""'
-    )
+        with schema_file_path.open("w") as f:
+            json.dump(json_schema, f)
 
-    shutil.rmtree(schema_dir)
+        subprocess.check_call(
+            [
+                "yarn",
+                "json2ts",
+                "-i",
+                str(schema_file_path),
+                "-o",
+                str(ts_output_path),
+                "--bannerComment",
+                "",
+            ]
+        )
 
-    if json2ts_exit_code == 0:
-        clean_output_file(output)
-    else:
-        raise RuntimeError(f'"{json2ts_cmd}" failed with exit code {json2ts_exit_code}.')
+        clean_output_file(ts_output_path)
 
 
 @click.command()
-@click.option("--output", default="index.d.ts", help="Output typescript file.")
-def cli(output):
-    main(output)
+@click.option(
+    "--output",
+    "ts_output_path",
+    type=Path,
+    default=Path("index.d.ts"),
+    help="Output typescript file.",
+)
+def cli(**kwargs):
+    main(**kwargs)
```

Comment:
What's the advantage of using kwargs here? My personal preference would be to be more explicit when specifying parameters to `cli()` and in calling `main()`, but is this preparing for a future capability or does it provide something else for us now?

Reply:
i just removed the cli() function and we call main directly now